lib/ClangImporter/ImportDecl.cpp

```diff
@@ -345,6 +345,23 @@ static bool isNSDictionaryMethod(const clang::ObjCMethodDecl *MD,
   return true;
 }
 
+static void forEachDistinctName(
+    ClangImporter::Implementation &impl, const clang::NamedDecl *decl,
+    llvm::function_ref<void(ImportedName, ImportNameVersion)> action) {
+  using ImportNameKey = std::pair<DeclName, EffectiveClangContext>;
+  SmallVector<ImportNameKey, 8> seenNames;
```

Comment:
Perhaps SmallSet? Yeah, std::set is not ideal when it grows outside the static size, but at least it has reasonable complexity when very large.

If you're opposed to SmallSet, then DeclName is pointer-like. For EffectiveClangContext, do you want the unresolved name here, or the Clang decl, or what are you expecting? An unresolved name might compare unequal to the same name but resolved to a Clang decl, for example.